src/components/shared/MarkerTooltipContents.js

```diff
@@ -34,6 +47,50 @@ function getMarkerDetails(data: MarkerPayload): React$Element<*> | null {
           </div>
         );
       }
+      case 'GCMinor': {
+        if (data.nursery && data.nursery.reason) {
```

Comment:
if the nursery property can be absent, you can mark it so, like this:
```
nursery ?: {|
  reason: string,
|}
```

If it's always present but can be undefined or null:
```
nursery: ?{|
  reason: string,
|}
```

If it's always present but can be null:
```
nursery: {|
  reason: string,
|} | null
```
We prefer the latter in our own code, but here of course we don't control (directly) what we get from the profiler -- I think in this case we actually want the structure to be as light as possible so the former is more likely. I think you know better than me how the declaration should be :)

With these declarations Flow will warn us in flow-checked code that you need to guard the code accessing `reason` with an if condition.

As you notice, the flow check is highly dependent on the correctness of the definitions. This is not unlilke rust unsafe code where you need to take care to the correctness of the interface :)


There was another question in your comment: could flow ensure the correctness of some data structure at runtime? The answer is: it can't. Because all flow information is erased at build time, and so nothing can be checked at run time. This is erased simply because it's not standard JavaScript and we need our lovely JS engine to be able to parse our code :)

Hope this helps !